### ISSUE TITLE
Errors pointed out by make pylint

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1729]
This is a reminder issue to fix some errors pointed out by `make pylint`.

Most are false positives, but a few are real errors.

In case someone else wants to run this - there was an issue (https://github.com/PyCQA/pylint/issues/2434), currently `make pylint` only works with `astroid` and `pylint` master.

```
$ make pylint
pylint -E gammapy/ \
	--ignore=_astropy_init.py,gammapy/extern \
	-d E0611,E1101,E1103 \
	--msg-template='{C}: {path}:{line}:{column}: {msg} ({symbol})' -f colorized
************* Module gammapy.__main__
E: gammapy/__main__.py:14:9: No value for argument 'log_level' in function call (no-value-for-parameter)
E: gammapy/__main__.py:14:9: No value for argument 'ignore_warnings' in function call (no-value-for-parameter)
************* Module gammapy.maps.hpx
E: gammapy/maps/hpx.py:9:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.maps.wcs
E: gammapy/maps/wcs.py:1086:8: Possible unbalanced tuple unpacking with sequence defined at line 1079: left side has 2 label(s), right side has 3 value(s) (unbalanced-tuple-unpacking)
************* Module gammapy.background.reflected
E: gammapy/background/reflected.py:62:57: bad operand type for unary -: tuple (invalid-unary-operand-type)
E: gammapy/background/reflected.py:62:57: bad operand type for unary -: NoneType (invalid-unary-operand-type)
************* Module gammapy.cube.models
E: gammapy/cube/models.py:184:22: self.spatial_model is not callable (not-callable)
E: gammapy/cube/models.py:185:23: self.spectral_model is not callable (not-callable)
************* Module gammapy.utils.docs
E: gammapy/utils/docs.py:58:25: Too many arguments for format string (too-many-format-args)
************* Module gammapy.utils.energy
E: gammapy/utils/energy.py:47:15: Argument 'dtype' passed by position and keyword in classmethod call (redundant-keyword-arg)
************* Module gammapy.utils.serialization.xml
E: gammapy/utils/serialization/xml.py:326:37: Undefined variable 'parname_registry' (undefined-variable)
************* Module gammapy.utils.serialization.tests.test_xml
E: gammapy/utils/serialization/tests/test_xml.py:149:4: Assigning to function call which doesn't return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:149:10: No value for argument 'filename' in method call (no-value-for-parameter)
E: gammapy/utils/serialization/tests/test_xml.py:165:4: Assigning to function call which doesn't return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:165:10: No value for argument 'filename' in method call (no-value-for-parameter)
************* Module gammapy.utils.distributions.general_random
E: gammapy/utils/distributions/general_random.py:4:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.image.plotting
E: gammapy/image/plotting.py:301:8: Unable to import 'skimage.color' (import-error)
************* Module gammapy.scripts.main
E: gammapy/scripts/main.py:73:4: No value for argument 'log_level' in function call (no-value-for-parameter)
E: gammapy/scripts/main.py:73:4: No value for argument 'ignore_warnings' in function call (no-value-for-parameter)
************* Module gammapy.irf.psf_gauss
E: gammapy/irf/psf_gauss.py:224:8: Assigning to function call which doesn't return (assignment-from-no-return)
************* Module gammapy.irf.psf_table
E: gammapy/irf/psf_table.py:542:12: function already defined line 521 (function-redefined)
************* Module gammapy.spectrum.observation
E: gammapy/spectrum/observation.py:6:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.spectrum.fit
E: gammapy/spectrum/fit.py:263:22: Unexpected keyword argument 'mu_on' in function call (unexpected-keyword-arg)
E: gammapy/spectrum/fit.py:263:22: No value for argument 'n_off' in function call (no-value-for-parameter)
E: gammapy/spectrum/fit.py:263:22: No value for argument 'alpha' in function call (no-value-for-parameter)
E: gammapy/spectrum/fit.py:263:22: No value for argument 'mu_sig' in function call (no-value-for-parameter)
************* Module gammapy.spectrum.energy_group
E: gammapy/spectrum/energy_group.py:21:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.astro.source.pulsar
E: gammapy/astro/source/pulsar.py:141:15: self.luminosity_spindown is not callable (not-callable)
************* Module gammapy.data.event_list
E: gammapy/data/event_list.py:1025:25: Value 'self' is unsubscriptable (unsubscriptable-object)
E: gammapy/data/event_list.py:1025:40: Value 'self' is unsubscriptable (unsubscriptable-object)
************* Module gammapy.data.observations
E: gammapy/data/observations.py:9:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.data.tests.test_pointing
E: gammapy/data/tests/test_pointing.py:18:4: Method should have "self" as first argument (no-self-argument)
************* Module gammapy.stats.feldman_cousins
E: gammapy/stats/feldman_cousins.py:6:0: Unable to import 'gammapy.extern.six.moves' (import-error)
************* Module gammapy.stats.fit_statistics
E: gammapy/stats/fit_statistics.py:161:17: bad operand type for unary -: list (invalid-unary-operand-type)
E: gammapy/stats/fit_statistics.py:168:17: bad operand type for unary -: list (invalid-unary-operand-type)
```

**Comment by cdeil** [comment]
This is a false positive, there is no issue / error:
```
************* Module gammapy.irf.psf_gauss
E: gammapy/irf/psf_gauss.py:224:8: Assigning to function call which doesn't return (assignment-from-no-return)
```
Reported here: https://github.com/PyCQA/pylint/issues/2435

**Comment by cdeil** [comment]
Most are fixed or disabled in #1732 . In cases of false positives I've filed issues with `pylint` and left a code comment in Gammapy pointing to that issue, so that we can clean this up further in the future.

The remaining errors are:
```
$ make pylint
pylint -E gammapy/ \
	--ignore=_astropy_init.py,gammapy/extern \
	-d E0611,E1101,E1103 \
	--msg-template='{C}: {path}:{line}:{column}: {msg} ({symbol})' -f colorized
************* Module gammapy.maps.wcs
E: gammapy/maps/wcs.py:1086:8: Possible unbalanced tuple unpacking with sequence defined at line 1079: left side has 2 label(s), right side has 3 value(s) (unbalanced-tuple-unpacking)
************* Module gammapy.utils.serialization.xml
E: gammapy/utils/serialization/xml.py:326:37: Undefined variable 'parname_registry' (undefined-variable)
************* Module gammapy.utils.serialization.tests.test_xml
E: gammapy/utils/serialization/tests/test_xml.py:149:4: Assigning to function call which doesn't return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:149:10: No value for argument 'filename' in method call (no-value-for-parameter)
E: gammapy/utils/serialization/tests/test_xml.py:165:4: Assigning to function call which doesn't return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:165:10: No value for argument 'filename' in method call (no-value-for-parameter)
************* Module gammapy.astro.source.pulsar
E: gammapy/astro/source/pulsar.py:141:15: self.luminosity_spindown is not callable (not-callable)
make: *** [pylint] Error 2
```

pylint can report a ton of more warnings and code style issues (but also a ton of false positives and irrelevant things):
https://gist.github.com/cdeil/08be5abbfe3626b8a9e7ab7fc399fbff

I'm moving this issue to v0.9, let's continue to improve code quality in the coming months ...

**Comment by cdeil** [comment]
I'm closing this old issue.

In #322 we now have this:
```
$ make pylint
pylint -E gammapy/ \
	--ignore=_astropy_init.py,gammapy/extern \
	-d E0611,E1101,E1103 \
	--msg-template='{C}: {path}:{line}:{column}: {msg} ({symbol})'
************* Module gammapy.utils.energy
E: gammapy/utils/energy.py:46:15: Argument 'dtype' passed by position and keyword in classmethod call (redundant-keyword-arg)
************* Module gammapy.utils.fitting.fit
E: gammapy/utils/fitting/fit.py:86:4: An attribute defined in gammapy.utils.fitting.fit line 83 hides this method (method-hidden)
************* Module gammapy.utils.serialization.xml
E: gammapy/utils/serialization/xml.py:330:37: Undefined variable 'parname_registry' (undefined-variable)
************* Module gammapy.utils.serialization.tests.test_xml
E: gammapy/utils/serialization/tests/test_xml.py:150:4: Assigning result of a function call, where the function has no return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:150:10: No value for argument 'filename' in method call (no-value-for-parameter)
E: gammapy/utils/serialization/tests/test_xml.py:168:4: Assigning result of a function call, where the function has no return (assignment-from-no-return)
E: gammapy/utils/serialization/tests/test_xml.py:168:10: No value for argument 'filename' in method call (no-value-for-parameter)
************* Module gammapy.spectrum.fit
E: gammapy/spectrum/fit.py:252:4: An attribute defined in gammapy.utils.fitting.fit line 83 hides this method (method-hidden)
make: *** [pylint] Error 2
```

So we have just a handful of remaining issues, and those should be discussed and fixed one by one in the coming months, IMO not in a "fix all pylint" issue thread any more.
